libyui-ncurses/src/NCWidget.h

```diff
@@ -31,6 +31,32 @@
 #include "NCurses.h"
 #include "tnode.h"
 
+// #pragma GCC diagnostic push
+//
+// This time (unlike in libyui/src/YMenuButton.h) not pushing/popping the
+// changed behavior since we really want it for all of libyui-ncurses and
+// libyui-ncurses-pkg.
+//
+// Notice that this does not affect any applications written with libyui since
+// everything below libyui-ncurses/src are not API classes or headers; those
+// are handled strictly within the UI plug-ins. Only those below libyui/src are
+// API classes and headers.
+
+#pragma GCC diagnostic ignored "-Woverloaded-virtual"
```

Comment:
I'm not a C++ expert, just a question: Would using the `override` keyword help to get rid of those warnings?  Could you try?

https://stackoverflow.com/a/18198377

Reply:
In this case IMHO no, since it's about a non-virtual function hiding a virtual one.

The tricky part is that if the signature is different, it won't replace (override) the virtual function AFAIK. That is what the message is all about.

Reply:
And even if the new function is virtual, if the signature (the parameters) are different, it won't override the virtual function of the base class; it just introduces a new virtual function that exists in parallel to the one of the base class. We seem to be doing that a lot in the Rest API.